BioinformaticTools-inator/BioinformaticToolsInator.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict, List, Tuple, Union
+from modules import dna_rna_funcs, fastq_funcs
+
+
+def run_dna_rna_tools(
+        *args: str
+        ) -> Union[List[str], Dict[str, Dict[str, int]], str]:
+    """
+    Performs a specified action on all sequences transmitted from input
+
+    Args:
+    - DNA or RNA sequences as str and a procedure to perform
+      as the last argument
+
+    Possible procedures:
+    - Main procedures: reverse, transcribe, complement, reverse_complement
+    - Additional procedures:
+      nucleotide_frequency, gc_content, find_start_codons
+
+    Returns:
+    - Union[List[str], Dict[str, Dict[str, int]], str]:
+      sequence or a collection of sequences with paramters
+    """
+    *seqs, operation = args
+    for seq in seqs:
+        dna_rna_funcs.is_valid_na(seq)
+    if operation == 'reverse':
+        return dna_rna_funcs.reverse(*seqs)
+    elif operation == 'transcribe':
+        return dna_rna_funcs.transcribe(*seqs)
+    elif operation == 'complement':
+        return dna_rna_funcs.complement(*seqs)
+    elif operation == 'reverse_complement':
+        return dna_rna_funcs.reverse_complement(*seqs)
+    elif operation == 'nucleotide_frequency':
+        return dna_rna_funcs.nucleotide_frequency(*seqs)
+    elif operation == 'gc_content':
+        return dna_rna_funcs.gc_content(*seqs)
+    elif operation == 'find_start_codons':
+        return dna_rna_funcs.find_start_codons(*seqs)
```

Comment:
Здесь есть повторяющийся паттерн. Конечно, этот код рабочий, но его можно сократить, сделав словарь, где ключи - названия операций, а значения - сами функции. Тогда в следующий раз при добавлении новой подфункции тебе не придется исправлять главную функцию, а можно будет просто добавить соотвествующий ключ и значение в словарь. Словарь же можно положить к остальным вспомогательным функциям. 

```suggestion
    if operation in fn_map.keys():
        fn = dna_rna_funcs.fn_map[operation] 
        return fn(*seqs)
        
        
        #Словарь должен выглядить как-то так: 
        
    fn_map: dict = {
    "reverse": reverse,
    "complement": complement,
    "function": function
}

```

Reply:
Не обязательно (Антону например больше нравится через услования) но, да, так можно. Как минимум полезно знать про разные варианты